cmd/config.go

```diff
@@ -274,12 +276,30 @@ type BeelineConfig struct {
 	Dataset string
 	// SampleRate is the (positive integer) denominator of the sample rate.
 	// Default: 1 (meaning all traces are sent). Set higher to send fewer traces.
-	SampleRate uint
+	SampleRate uint32
 	// Mute disables honeycomb entirely; useful in test environments.
 	Mute bool
 	// Many other fields of beeline.Config are omitted as they are not yet used.
 }
 
+// makeSampler constructs a SamplerHook which will deterministically decide if
+// any given span should be sampled based on its TraceID, which is shared by all
+// spans within a trace. A sample rate of 0 defaults to a sample rate of 1
+// (i.e. all events are sent).
+func makeSampler(rate uint32) func(fields map[string]interface{}) (bool, int) {
+	if rate == 0 {
+		rate = 1
+	}
+	upperBound := math.MaxUint32 / rate
+
+	return func(fields map[string]interface{}) (bool, int) {
+		h := sha1.Sum([]byte(fields["trace.trace_id"].(string)))
```

Comment:
I'd prefer doing the `, ok` notation for this type assertion, and return some reasonable default (always true?) in case we have some surprising bug where `trace.trace_id` is unset or not a string.

Reply:
Done.